### PR TITLE
feat(slack): add test_connection method

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Context

This PR adds functionality to check if the Slack bot can authenticate and access a specific Slack channel (public or private) before attempting to send a message. It prevents errors caused by invalid tokens or inaccessible channels.

### Description

- Added a static test_connection method to the Slack class.
- The method validates the Slack token using auth_test and checks channel access via conversations_list.
- Handles both public and private channels.
- Included unit tests with unittest.mock to simulate Slack API responses and ensure the method behaves as expected.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jfagoagas — APPROVED** [review]
discarded approve.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
As we did with the Providers, we need to create Custom Exceptions and mimic what we are doing in the `test_connection` for them with the exceptions we create and with the `raise_on_exception` boolean. Please take a look at it.

Thanks!

**Comment by MrCloudSec** [comment]
> raise_on_exception

Done!

File: prowler/lib/outputs/slack/slack.py [hunk 1]
```diff
@@ -230,3 +231,48 @@ def __create_title__(self, identity, stats) -> str:
...
+            channel (str): The Slack channel to be validated.
+
+        Returns:
+            Connection: A Connection object.
+        """
+        try:
+            client = WebClient(token=token)
+            # Test if the token is valid
+            auth_response = client.auth_test()
+            if auth_response["ok"]:
```
Comment (jfagoagas) [inline prowler/lib/outputs/slack/slack.py:260]: We need to add a custom exception here for an invalid token.

File: prowler/lib/outputs/slack/slack.py [hunk 2]
```diff
@@ -230,3 +231,48 @@ def __create_title__(self, identity, stats) -> str:
...
+                # Test if the channel is accessible
+                channels_response = client.conversations_list(
+                    types="public_channel", limit=1000
```
Comment (jfagoagas) [inline prowler/lib/outputs/slack/slack.py]: Do we need the `limit=1000`? Could it be 1?

File: prowler/lib/outputs/slack/slack.py [hunk 3]
```diff
@@ -230,3 +231,48 @@ def __create_title__(self, identity, stats) -> str:
...
+                )
+                if channels_response["ok"]:
```
Comment (jfagoagas) [inline prowler/lib/outputs/slack/slack.py:265]: We need to add a custom exception here for a not accessible public channel.

File: prowler/lib/outputs/slack/slack.py [hunk 4]
```diff
@@ -230,3 +231,48 @@ def __create_title__(self, identity, stats) -> str:
...
+                    for ch in channels_response["channels"]:
+                        if ch["name"] == channel or ch["id"] == channel:
+                            return Connection(is_connected=True)
+                private_channels_response = client.conversations_list(
```
Comment (jfagoagas) [inline prowler/lib/outputs/slack/slack.py]: We need to add a custom exception here for a not accessible private channel.

File: prowler/lib/outputs/slack/slack.py [hunk 5]
```diff
@@ -230,3 +231,48 @@ def __create_title__(self, identity, stats) -> str:
...
+                    types="private_channel", limit=1000
```
Comment (jfagoagas) [inline prowler/lib/outputs/slack/slack.py]: Do we need the `limit=1000`? Could it be 1?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/lib/outputs/slack/slack.py [hunk 6]
```diff
@@ -230,3 +236,81 @@ def __create_title__(self, identity, stats) -> str:
...
+                        error=SlackChannelNotFound(
+                            file=os.path.basename(__file__),
+                            message=(
+                                channels_response["error"]
+                                if "error" in channels_response
+                                else "Unknown error"
+                            ),
+                        )
+                    )
+            else:
```
Comment (jfagoagas) [inline prowler/lib/outputs/slack/slack.py]: you can create the exception within the `else` block and then `raise` it or include it within `Connection`.

File: prowler/lib/outputs/slack/slack.py [hunk 7]
```diff
@@ -230,3 +236,81 @@ def __create_title__(self, identity, stats) -> str:
...
+            auth_response = client.auth_test()
+            if auth_response["ok"]:
+                # Test if the channel is accessible
+                channels_response = client.conversations_info(
+                    token=token, channel=channel
+                )
+                if channels_response["ok"]:
+                    return Connection(is_connected=True)
+                else:
+                    if raise_on_exception:
```
Comment (jfagoagas) [inline prowler/lib/outputs/slack/slack.py]: you can create the exception within the `else` block and then `raise` it or include it within `Connection`.

File: prowler/lib/outputs/slack/slack.py [hunk 8]
```diff
@@ -230,3 +236,81 @@ def __create_title__(self, identity, stats) -> str:
...
+                    )
+                )
+
+        except Exception as error:
+            logger.error(
+                f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+            if raise_on_exception:
+                raise SlackClientError(
+                    file=os.path.basename(__file__),
```
Comment (jfagoagas) [inline prowler/lib/outputs/slack/slack.py:298]: Please, add `original_exception` here.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Please review my comments and add the missing tests for the not covered lines. Thanks!

File: prowler/lib/outputs/slack/slack.py (hunk 7 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/slack/slack.py]: Sure!

File: prowler/lib/outputs/slack/slack.py (hunk 6 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/slack/slack.py]: Sure!

File: prowler/lib/outputs/slack/slack.py (hunk 8 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/slack/slack.py:298]: Sure!

File: prowler/lib/outputs/slack/slack.py (hunk 6 above)
Comment (jfagoagas) [reply prowler/lib/outputs/slack/slack.py]: Thanks!

File: prowler/lib/outputs/slack/slack.py (hunk 8 above)
Comment (jfagoagas) [reply prowler/lib/outputs/slack/slack.py:298]: Thanks!

File: prowler/lib/outputs/slack/slack.py (hunk 7 above)
Comment (jfagoagas) [reply prowler/lib/outputs/slack/slack.py]: Thanks!

**Review by jfagoagas — APPROVED** [review]
🚀 🚀 🚀 🚀 🚀 🚀

**Pull request merged**